### ISSUE TITLE
_identifiers endpoint returns 404 for a given node

### ISSUE THREAD

**Issue by sersut** [chef/chef-server#280]
We have an issue in reporting where we call `/organizations/~s/nodes/~s/_identifiers` endpoint to get the authz_id for a node and get 404. 

Looked into it with @sdelano and looks like https://github.com/chef/chef-server/commit/ff27f1bb73618f4a650115d7ad4af6597fb75b42#diff-30fe53b1d1dd48b17b5d39481f04c894 caused the issue.

```
2015-05-20T23:05:53Z erchef@127.0.0.1 method=GET; path=/organizations/dev-org/nodes/client_node/_identifiers; status=404; req_id=g3IAA2QAEGVyY2hlZkAxMjcuMC4wLjEBAAA3oQAAAAAAAAAA; org_name=dev-org; msg={not_found,<<"nodes">>,<<"client_node">>}; couchdb_groups=false; couchdb_organizations=false; couchdb_containers=false; couchdb_acls=false; 503_mode=false; couchdb_associations=false; couchdb_association_requests=false; req_time=3; rdbms_time=0; rdbms_count=1; user=client_node; req_api_version=0;
```

```
(erchef@127.0.0.1)1> redbug:start("chef_wm_object_identifiers:do_node_fetch/4", [{msgs,10},{time, 60000}]).
{631,1}

% 23:20:33 <0.2242.0>({mochiweb_acceptor,init,3})
% chef_wm_object_identifiers:do_node_fetch(<<"client_node">>, undefined, {object_identifier_state,undefined,undefined,undefined}, {context,0,<<"g3IAA2QAEGVyY2hlZkAxMjcuMC4wLjEBAABfbgAAAAAAAAAA">>,
         {server,"127.0.0.1",5987,[],[]},
         {xdarklaunch,undefined,
                      <<"503_mode=0;couchdb_association_requests=0;couchdb_acls=0;couchdb_organizations=0;couchdb_containers=0;couchdb_groups=0;couchdb_associations=0;">>,
                      {dict,7,16,16,8,80,48,
                            {[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[]},
                            {{[],
                              [[<<"couchdb_groups">>|false]],
                              [[<<"couchdb_organizations">>|false]],
                              [[<<"couchdb_containers">>|false]],
                              [[<<"couchdb_acls">>|false]],
                              [],
                              [[<<"503_mode">>|false]],
                              [],[],
                              [[<<"couchdb_association_requests">>|false],
                               [<<"couchdb_associations">>|false]],
                              [],[],[],[],[],[]}}}}})
```

**Comment by marcparadise** [comment]
I'm on the road at the moment, but a couple of things: 

Most importantly:  If we need a quick fix, restore the old code except use: 

```{OrgId, _} chef_db:fetch_org_metadata(Ctx, OrgName) ``` 

However, I'm confused on one point: the dev-org is part of hte URL, and if the name has been resolved as part of the request prior to this point, so should the guid - unless resource_exists is called before malformed (which is where this gets set). 

Will check in when I get back.

**Comment by stevendanna** [comment]
resource_exists is called after malformed_request according to this flow chart:

https://raw.githubusercontent.com/wiki/basho/webmachine/images/http-headers-status-v3.png

However, the chef_wm_object_identifiers module doesn't mixin malfromed_request, so it is likely never being called.

**Comment by stevendanna** [comment]
Tests: https://github.com/chef/chef-server/pull/284
Proposed Fix: https://github.com/chef/chef-server/pull/286

**Comment by stevendanna** [comment]
This fix has been released to Hosted and will be in the 12.1 release.  Since the breakage has not yet been released in a package, I'm going to close this out now.
